cadquery/occ_impl/exporters/dxf.py

```diff
@@ -1,154 +1,331 @@
+        setup: Union[bool, List[str]] = False,
+        doc_units: int = units.MM,
+        *,
+        metadata: Union[Dict[str, str], None] = None,
+        approx: Optional[ApproxOptions] = None,
+        tolerance: float = 1e-3,
+    ):
+        """Initialize DXF document.
+
+        :param dxfversion: :attr:`DXF version specifier <ezdxf-stable:ezdxf.document.Drawing.dxfversion>`
+            as string, default is "AC1027" respectively "R2013"
+        :param setup: setup default styles, ``False`` for no setup, ``True`` to set up
+            everything or a list of topics as strings, e.g. ``["linetypes", "styles"]``
+            refer to :func:`ezdxf-stable:ezdxf.new`.
+        :param doc_units: ezdxf document/modelspace :doc:`units <ezdxf-stable:concepts/units>`
+        :param metadata: document :ref:`metadata <ezdxf-stable:ezdxf_metadata>` a dictionary of name value pairs
+        :param approx: Approximation strategy for converting :class:`cadquery.Workplane` objects to DXF entities:
+
+            ``None``
+                no approximation applied
+            ``"spline"``
+                all splines approximated as cubic splines
+            ``"arc"``
+                all curves approximated as arcs and straight segments
+
+        :param tolerance: Approximation tolerance for converting :class:`cadquery.Workplane` objects to DXF entities.
+        """
+        if metadata is None:
+            metadata = {}
+
+        self._DISPATCH_MAP = {
+            "LINE": self._dxf_line,
+            "CIRCLE": self._dxf_circle,
+            "ELLIPSE": self._dxf_ellipse,
+        }
+
+        self.approx = approx
+        self.tolerance = tolerance
+
+        self.document = ezdxf.new(dxfversion=dxfversion, setup=setup, units=doc_units)  # type: ignore[attr-defined]
+        self.msp = self.document.modelspace()
+
+        doc_metadata = self.document.ezdxf_metadata()
+        for key, value in metadata.items():
+            doc_metadata[key] = value
+
+    def add_layer(
+        self, name: str, *, color: int = 1, linetype: str = "CONTINUOUS"
+    ) -> Self:
+        """Create a layer definition
+
+        Refer to :ref:`ezdxf layers <ezdxf-stable:layer_concept>` and
+        :doc:`ezdxf layer tutorial <ezdxf-stable:tutorials/layers>`.
+
+        :param name: layer definition name
+        :param color: ezdxf color
+        :param linetype: ezdxf :doc:`line type <ezdxf-stable:concepts/linetypes>`
+        """
+        self.document.layers.add(name, color=color, linetype=linetype)
+
+        return self
+
+    def add_shape(self, workplane: Workplane, layer: str = "") -> Self:
+        """Add CadQuery shape to a DXF layer.
+
+        :param workplane: CadQuery Workplane
+        :param layer: layer definition name
+        """
+        plane = workplane.plane
+        shape = toCompound(workplane).transformShape(plane.fG)
+
+        general_attributes = {}
+        if layer:
+            general_attributes["layer"] = layer
```

Comment:
Same here: not covered

https://app.codecov.io/gh/CadQuery/cadquery/pull/1267/blob/cadquery/occ_impl/exporters/dxf.py